src/new_storage/storage_manager/handle.rs

```diff
@@ -414,80 +338,272 @@ impl FileHandleInner {
+                (content, version)
+            };
+
+            task.backend.write(&path, &content).await?;
+            {
+                let mut block = task.block.write();
+                // Check version
+                if block.version() != version {
+                    warn!(
+                        "Version mismatch previous: {}, current: {}",
+                        version,
+                        block.version()
+                    );
+                    continue;
+                }
+                block.set_dirty(false);
+            }
+            {
+                task.cache.lock().unpin(&CacheKey {
+                    ino: self.ino,
+                    block_id: task.block_id,
+                });
+                break;
             }
         }
 
-        rx.await
-            .unwrap_or_else(|_| panic!("The sender should not be closed."))
-            .map_or(Ok(()), Err)
+        Ok(task.file_size)
+    }
+
+    /// Write the blocks to the backend storage system concurrently.
+    async fn write_blocks<M: MetaData + Send + Sync + 'static>(
+        self: Arc<Self>,
+        tasks: &Vec<Arc<WriteTask>>,
+        metadata_client: Arc<M>,
+    ) -> Option<StorageError> {
+        let mut handles = Vec::new();
+        let mut result = None;
+        let mut done_file_size = None;
+        for task in tasks {
+            let arc_self = Arc::clone(&self);
+            let handle = tokio::spawn(arc_self.write_back_block(Arc::clone(task)));
+            handles.push(handle);
+        }
+        // Make sure current blocks is finished.
+        for handle in handles {
+            match handle.await {
+                Err(e) => {
+                    result = Some(StorageError::Internal(e.into()));
+                }
+                Ok(Err(e)) => {
+                    result = Some(e);
+                }
+                Ok(Ok(file_size)) => {
+                    self.dirty_count
+                        .fetch_sub(1, std::sync::atomic::Ordering::SeqCst);
+                    done_file_size = Some(file_size);
+                }
+            }
+        }
+
+        // Write back the file size to the meta data server.
+        if let Some(file_size) = done_file_size {
+            // Commit current metadata to the meta data server.
+            let ino = self.ino;
+            info!(
+                "Commit meta data for ino: {} with attr size: {:?} current dirty_size: {:?}",
+                ino,
+                file_size,
+                self.attr.read().await.dirty_filesize
+            );
+            if let Err(e) = metadata_client
+                .write_remote_size_helper(ino, file_size)
+                .await
+            {
+                error!("Failed to commit meta data, the error is {e}.");
+            }
+
+            let mut attr_write = self.attr.write().await;
+            if let Some(dirty_filesize) = attr_write.dirty_filesize {
+                if dirty_filesize <= file_size {
```

Comment:
I think should be `if dirty_filesize == file_size`. 
Set self.attr directly is not bulletproof,  should use latest attr query from  write_remote_size_helper after update suc

Reply:
OK, I have updated the filesize with remote size.